tests/lib/tools/tests.invariant

```diff
@@ -122,6 +123,22 @@ check_leftover_defer_sh() {
 	fi
 }
 
+check_broken_snaps() {
+	n="$1" # invariant name
+	(
+		# awk is hard - this just gives us the names of snaps that are broken
+		for brokesn in $(snap list --all | tail -n +2 | awk '{print $(NF) " " $1}' | grep 'broken ' | awk '{print $(NF)}' | sort | uniq); do
```

Comment:
Hmm did you mean something like:
```
# fist column is the snap name, revision is 3rd from the end, we cannot use $3 as
# broken snaps have empty version column
snap list --all | awk '/,?broken,?/ { print $1, $(NF-3) }' | while read -r name rev; do 
    echo "snap $name ($rev) is broken"
done
```
With the dump from debug log we get:
```
snap core18 (1888) is broken
snap pi (74) is broken
snap snapd (8791) is broken
```

On a side note, perhaps we should really print some placeholder for a version if the snap is broken. We use `-` for the tracking and publisher columns when the snap is unasserted. cc @pedronis

Reply:
Added a placeholder in https://github.com/snapcore/snapd/pull/11166

Reply:
It was indeed my intention to have the revision number also printed but since I couldn't figure out a nice way to get that due to the missing version in the second column I resorted to this hackery but your hackery is more useful as it also has the revision number :-) Thanks for the suggestion

Reply:
Added this suggestion since the other PR has not yet landed

Reply:
Hmm, I added this suggestion but now the tests are not failing, I think there is something wrong with this somewhere... Feel free to push things to this branch to try other things, or we could just wait until that PR has landed to do the easy and simple implementation

Reply:
followup in https://github.com/snapcore/snapd/pull/11204